src/github/utils.ts

```diff
@@ -171,50 +171,129 @@ export function parseGraphQLComment(comment: ReviewComment): Comment {
 		let type = convertGraphQLEventType(event.__typename);
-		event.event = type;
-
-		if (event.event === EventType.Commented) {
-			event.canEdit = event.viewerCanUpdate;
-			event.canDelete = event.viewerCanDelete;
-			event.user = event.author;
-			event.id = event.databaseId;
-			event.htmlUrl = event.url;
-		}
 
-		if (event.event === EventType.Reviewed) {
-			event.user = event.author;
-			event.canEdit = event.viewerCanUpdate;
-			event.canDelete = event.viewerCanDelete;
-			event.id = event.databaseId;
-			event.htmlUrl = event.url;
-			event.submittedAt = event.submittedAt;
-		}
+		switch (type) {
+			case Common.EventType.Commented:
+				let commentEvent = event as GraphQL.IssueComment;
+				ret.push({
+					htmlUrl: commentEvent.url,
+					body: commentEvent.body,
+					bodyHTML: commentEvent.bodyHTML,
+					user: commentEvent.author,
+					event: type,
+					canEdit: commentEvent.viewerCanUpdate,
+					canDelete: commentEvent.viewerCanDelete,
+					id: commentEvent.databaseId,
+					createdAt: commentEvent.createdAt
+				} as Common.CommentEvent);
+				return;
+			case Common.EventType.Reviewed:
+				let reviewEvent = event as GraphQL.Review;
+				ret.push({
+					event: type,
+					comments: [],
+					submittedAt: reviewEvent.submittedAt,
+					body: reviewEvent.body,
+					bodyHTML: reviewEvent.bodyHTML,
+					htmlUrl: reviewEvent.url,
+					user: reviewEvent.author,
+					authorAssociation: reviewEvent.authorAssociation,
+					state: reviewEvent.state,
+					id: reviewEvent.databaseId,
+				} as Common.ReviewEvent);
+				return;
+			case Common.EventType.Committed:
+				let commitEv = event as GraphQL.Commit;
+				ret.push({
+					event: type,
+					sha: commitEv.oid,
+					author: commitEv.author.user || { login: commitEv.committer.name, avatarUrl: commitEv.committer.avatarUrl },
+					htmlUrl: commitEv.url,
+					message: commitEv.message
+				} as Common.CommitEvent);
+				return;
+			case Common.EventType.Merged:
+				let mergeEv = event as GraphQL.MergedEvent;
+
+				ret.push({
+					event: type,
+					user: mergeEv.actor,
+					createdAt: mergeEv.createdAt,
+					mergeRef: mergeEv.mergeRef.name,
+					sha: mergeEv.commit.oid,
+					commitUrl: mergeEv.commit.commitUrl,
+					url: mergeEv.url,
+					graphNodeId: mergeEv.id
+				} as Common.MergedEvent);
+				return;
+			case Common.EventType.Assigned:
+				let assignEv = event as GraphQL.AssignedEvent;
 
-		if (event.event === EventType.Committed) {
-			event.sha = event.oid;
-			event.author = event.author.user || { login: event.committer.name, avatarUrl: event.committer.avatarUrl };
-			event.htmlUrl = event.url;
+				ret.push({
+					event: type,
+					user: assignEv.user,
+					actor: assignEv.actor
+				} as Common.AssignEvent);
+				return;
+			default:
+				break;
 		}
 	});
 
-	return events;
+	return ret;
 }
 
-export function convertRESTTimelineEvents(events: any[]): TimelineEvent[] {
+export function convertRESTTimelineEvents(events: any[]): Common.TimelineEvent[] {

```

Comment:
if possible let's also replace `any` with a real type here, `avatar_url` isn't being converted to `avatarUrl`